### DISCUSSION TITLE
Any resource about the implementation of `SortedSet`?

### DISCUSSION THREAD

**Discussion by Jason5Lee** [dotnet/runtime#109864] [Q&A] [answered]
I have recently been reading some code related to data structures to learn about their implementations. I discovered that `SortedSet` is a top-down red-black tree implementation and that nodes do not need to maintain parent nodes, which is not common as I know. Could you recommend any resources where I can learn more about how it works?

**Comment by lilinus** [comment, marked as answer]
The source code for SortedSet has a couple of references to "Introduction to algorithms" by Thomas H. Cormen.
